humility-core/src/net.rs

```diff
@@ -2,7 +2,47 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at https://mozilla.org/MPL/2.0/.
 
-use anyhow::{bail, Result};
+use anyhow::{bail, Context, Result};
+use std::{fmt, net::Ipv6Addr, str::FromStr};
+
+#[derive(Copy, Clone, Eq, PartialEq)]
```

Comment:
Please add a doc comment here (doesn't have to be super long, but, what is this thing and why would I reach for it).